doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-ESP8266/index.md

```diff
@@ -5,4 +5,383 @@ permalink: /:path/
+
+## Connecting ESP8266
+
+To flash ESP8266 chip you need to connect it to your PC. This may differ depending on module you use.
+This guide covers instructions for ESP8266-01 and NodeMCU modules.
+
+### ESP8266-01
+
+To connect the ESP8266-01 to PC, a 3.3V USB-to-TTL connector is required.
+There are two boot modes for ESP8266: flash mode and run mode.
+The table below summarizes wiring scheme for both boot modes.
+
+<ul class="nav nav-tabs">
+    <li class="active"><a data-toggle="tab" href="#Flash">Fash Mode</a></li>
+    <li><a data-toggle="tab" href="#Run">Run mode</a></li>
+</ul>
+
+<div class="tab-content">
+<div id="Flash" class="tab-pane fade in active" markdown="1" >
+
+#### Flash mode
+
+| ESP8266-01 | USB-to-TTL |
+|:----------:|:----------:|
+| URXD       | TXD        |
+| UTXD       | RXD        |
+| CH_PD      | 3.3V       |
+| GND        | GND        |
+| VCC        | 3.3V       |
+| GPIO0      | GND        |
+| GPIO2      | 3.3V       |
+
+</div><div id="Run" class="tab-pane fade" markdown="1" >
+
+#### Run mode
+
+| ESP8266-01 | USB-to-TTL |
+|:----------:|:----------:|
+| URXD       | TXD        |
+| UTXD       | RXD        |
+| CH_PD      | 3.3V       |
+| GND        | GND        |
+| VCC        | 3.3V       |
+
+</div></div>
+
+### NodeMCU
+
+Connecting NodeMCU is much simpler -- just connect it via micro-USB cable.
+
+## Installing requirements
+
+Prior to developing Kaa applications for ESP8266 some requirements should be installed.
+The detailed installation instructions can be found below.
+
+1. Prerequisites
+
+        sudo apt-get install autoconf libtool libtool-bin bison build-essential gawk git gperf flex texinfo libtool libncurses5-dev libc6-dev-amd64 python-serial libexpat-dev python-setuptools
+
+2. Set the `ESPRESSIF_HOME` variable.
+This variable will be used throughout installation process, and denotes a directory where ESP8266 SDK and toolchain will be placed.
+You are free to set it to whatever you like.
+
+            export ESPRESSIF_HOME=/opt/Espressif/
+
+3. Install toolchain
+
+        cd $ESPRESSIF_HOME
+        git clone -b lx106 git://github.com/jcmvbkbc/crosstool-NG.git
+        cd crosstool-NG
+        ./bootstrap && ./configure --prefix=`pwd` && make && sudo make install
+        ./ct-ng xtensa-lx106-elf
+        ./ct-ng build
+
+4. Add path to toolchain binaries to your .bashrc:
+
+        echo "export PATH=\$PATH:$ESPRESSIF_HOME/crosstool-NG/builds/xtensa-lx106-elf/bin" >> ~/.bashrc
+
+4. Install ESP8266 RTOS SDK
+
+        cd $ESPRESSIF_HOME
+        export ESP_SDK_HOME=$ESPRESSIF_HOME/esp-rtos-sdk
+        git clone https://github.com/espressif/esp_iot_rtos_sdk.git $ESP_SDK_HOME
+        cd $ESP_SDK_HOME
+        git reset --hard 169a436ce10155015d056eab80345447bfdfade5
+        wget -O lib/libhal.a https://github.com/esp8266/esp8266-wiki/raw/master/libs/libhal.a
+        cd $ESP_SDK_HOME/include/lwip/arch
+        sed -i "s/#include \"c_types.h\"/\/\/#include \"c_types.h\"/" cc.h
```

Comment:
Don't do `cd` just to take a file. Use `sed -i "s/#include \"c_types.h\"/\/\/#include \"c_types.h\"/" $ESP_SDK_HOME/include/lwip/arch/cc.h`